Xamarin.Forms.Core/BindableLayout.cs

```diff
@@ -0,0 +1,261 @@
+﻿using System.Collections;
+using System.Collections.Specialized;
+
+namespace Xamarin.Forms.Core
+{
+	public static class BindableLayout
+	{
+		public static readonly BindableProperty ItemsSourceProperty =
+			BindableProperty.Create("ItemsSource", typeof(IList), typeof(Layout<View>), default(IList),
+				propertyChanged: (b, o, n) => { GetController(b).ItemsSource = (IList)n; });
+
+		public static readonly BindableProperty ItemTemplateProperty =
+			BindableProperty.Create("ItemTemplate", typeof(DataTemplate), typeof(Layout<View>),
+				propertyChanged: (b, o, n) => { GetController(b).ItemTemplate = (DataTemplate)n; });
+
+		public static readonly BindableProperty ItemTemplateSelectorProperty =
+			BindableProperty.Create("ItemTemplateSelector", typeof(DataTemplateSelector), typeof(Layout<View>),
+				propertyChanged: (b, o, n) => { GetController(b).ItemTemplateSelector = (DataTemplateSelector)n; });
+
+		public static readonly BindableProperty ControllerProperty =
+			 BindableProperty.Create("Controller", typeof(BindableLayoutController), typeof(Layout<View>), defaultValueCreator: (b) => new BindableLayoutController((Layout<View>)b),
+				 propertyChanged: (b, o, n) => OnControllerChanged(b, (BindableLayoutController)o, (BindableLayoutController)n));
+
+		public static void SetItemsSource(BindableObject b, IList value)
+		{
+			b.SetValue(ItemsSourceProperty, value);
+		}
+
+		public static IList GetItemsSource(BindableObject b)
+		{
+			return (IList)b.GetValue(ItemsSourceProperty);
+		}
+
+		public static void SetItemTemplate(BindableObject b, DataTemplate value)
+		{
+			b.SetValue(ItemTemplateProperty, value);
+		}
+
+		public static DataTemplate GetItemTemplate(BindableObject b)
+		{
+			return (DataTemplate)b.GetValue(ItemTemplateProperty);
+		}
+
+		public static void SetItemTemplateSelector(BindableObject b, DataTemplateSelector value)
+		{
+			b.SetValue(ItemTemplateSelectorProperty, value);
+		}
+
+		public static DataTemplateSelector GetItemTemplateSelector(BindableObject b)
+		{
+			return (DataTemplateSelector)b.GetValue(ItemTemplateSelectorProperty);
+		}
+
+		public static BindableLayoutController GetController(BindableObject b)
```

Comment:
shouldn't be `public`

Reply:
Done, I removed public. I think it's a missed opportunity